core/domain/state_domain.py

```diff
@@ -564,6 +620,14 @@ def get_all_html_content_strings(self):
         if self.solution:
             solution_html = self.solution.explanation.html
             html_list = html_list + [solution_html]
+            if self.solution.correct_answer:
+                interaction = (
+                    interaction_registry.Registry.get_interaction_by_id(
+                        self.id).to_dict())
+                if (interaction['answer_type'] ==
```

Comment:
Should we use `extensions/interactions/html_field_types_to_rule_specs.json` for this one as well? Like you can do interaction.can_have_solution if yes then if the anwer_type corresponds to anyof the `html_field_types_to_rule_specs` then follow the required pattern to extract it? (not sure) Basically this will open the ground for anty interaction to enable solution without making any changes here.

Reply:
done